sdk/python/kfp/components/_components.py

```diff
@@ -202,12 +203,20 @@ def _create_task_factory_from_component_spec(component_spec:ComponentSpec, compo
 
     def create_task_from_component_and_arguments(pythonic_arguments):
         #Converting the argument names and not passing None arguments
-        valid_argument_types = (str, int, float, bool, GraphInputArgument, TaskOutputArgument) #Hack for passed PipelineParams. TODO: Remove the hack once they're no longer passed here.
+        valid_argument_types = (str, int, float, bool, GraphInputArgument, TaskOutputArgument, PipelineParam) #Hack for passed PipelineParams. TODO: Remove the hack once they're no longer passed here.
         arguments = {
             pythonic_name_to_input_name[k]: (v if isinstance(v, valid_argument_types) else str(v))
             for k, v in pythonic_arguments.items()
             if v is not None
         }
+        for key in arguments:
+            if isinstance(arguments[key], PipelineParam):
+                for input_spec in component_spec.inputs:
+                    if input_spec.name == key:
+                        if arguments[key].param_type is not None and not check_types(input_spec.type, arguments[key].param_type.serialize()):
```

Comment:
Do we need `.serialize()` here? AFAIK, `check_types` already does everything.

Reply:
Yes because check_types function only accepts BaseType, str or dict not TypeMeta.

Reply:
I see.

Reply:
BTW, I'm adding a lot of unit tests to cover corner cases.